client/features/authSlice.tsx

```diff
@@ -0,0 +1,30 @@
+import { createSlice } from '@reduxjs/toolkit'
+import type { PayloadAction } from '@reduxjs/toolkit'
+import type { User } from '../../src/types/custom'
+import type { RootState } from '../app/store'
+
+type AuthState = {
+  user: User | null
+  token: string | null
+}
+
+const authSlice = createSlice({
+  name: 'auth',
+  initialState: { user: null, token: null } as AuthState,
+  reducers: {
+    setCredentials: (
+      state,
+      { payload: { user, token } }: PayloadAction<{ user: User; token: string }>
+    ) => {
+      state.user = user
+      state.token = token
+    },
+  },
```

Comment:
If you are using extraReducers on the api, you shouldn't need to use this action. I would recommend just adding a logout reducer that sets both to null - or not adding any reducers until you're ready to fully implement the logout feature